cli/src/test.js

```diff
@@ -10,100 +7,119 @@ import {
   runDockerContainer,
 } from './execDocker/docker.js';
 import { readIDappConfig } from './utils/idappConfigFile.js';
-import { TEST_INPUT_DIR, TEST_OUTPUT_DIR } from './config/config.js';
-
-const execAsync = util.promisify(exec);
+import {
+  IEXEC_WORKER_HEAP_SIZE,
+  TEST_INPUT_DIR,
+  TEST_OUTPUT_DIR,
+} from './config/config.js';
+import { handleCliError } from './utils/cli-helpers.js';
 
 export async function test(argv) {
-  await cleanTestOutput();
-  if (argv.docker) {
-    await testWithDocker(argv.params);
-  } else {
-    await testWithoutDocker(argv.params);
+  const spinner = ora();
+  try {
+    await cleanTestOutput({ spinner });
+    await testApp({ args: argv.params, spinner });
+    // TODO check output files
+    // - output required files
+    // - outpout dir size
+    await askShowTestOutput({ spinner });
```

Comment:
Should maybe not propose to see the results if app exited with an error?

Reply:
IMO it's good to know what was produced or not